src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -0,0 +1,1068 @@
+        try {
+            out.append(toHighHexDigit(value));
+            out.append(toLowHexDigit(value));
+            return out;
+        } catch (IOException ioe) {
+            throw new UncheckedIOException(ioe.getMessage(), ioe);
+        }
+    }
+
+    /**
+     * Returns the two hexadecimal characters for the {@code byte} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value a byte value
+     * @return the two hexadecimal characters for the byte value
+     */
+    public String toHexDigits(byte value) {
+        byte[] rep = new byte[2];
+        rep[0] = (byte)toHighHexDigit(value);
+        rep[1] = (byte)toLowHexDigit(value);
+        try {
+            return jla.newStringNoRepl(rep, StandardCharsets.ISO_8859_1);
+        } catch (CharacterCodingException cce) {
+            throw new AssertionError(cce);
+        }
+    }
+
+    /**
+     * Returns the four hexadecimal characters for the {@code char} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value a {@code char} value
+     * @return the four hexadecimal characters for the {@code char} value
+     */
+    public String toHexDigits(char value) {
+        return toHexDigits((short)value);
+    }
+
+    /**
+     * Returns the four hexadecimal characters for the {@code short} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value a {@code short} value
+     * @return the four hexadecimal characters for the {@code short} value
+     */
+    public String toHexDigits(short value) {
+        byte[] rep = new byte[4];
+        rep[0] = (byte)toHighHexDigit((byte)(value >> 8));
+        rep[1] = (byte)toLowHexDigit((byte)(value >> 8));
+        rep[2] = (byte)toHighHexDigit((byte)value);
+        rep[3] = (byte)toLowHexDigit((byte)value);
+
+        try {
+            return jla.newStringNoRepl(rep, StandardCharsets.ISO_8859_1);
+        } catch (CharacterCodingException cce) {
+            throw new AssertionError(cce);
+        }
+    }
+
+    /**
+     * Returns the eight hexadecimal characters for the {@code int} value.
+     * Each nibble (4 bits) from most significant to least significant of the value
+     * is formatted as if by {@link #toLowHexDigit(int) toLowHexDigit(nibble)}.
+     * The delimiter, prefix and suffix are not used.
+     *
+     * @param value an {@code int} value
+     * @return the eight hexadecimal characters for the {@code int} value
+     */
```

Comment:
I recommend added an @see link to Integer.toHexString (and Long.toHexString for the long overload).